src/core/components/libp2p.js

```diff
@@ -15,27 +15,43 @@ module.exports = function libp2p (self) {
           return callback(err)
         }
 
-        const options = {
-          mdns: get(config, 'Discovery.MDNS.Enabled'),
-          webRTCStar: get(config, 'Discovery.webRTCStar.Enabled'),
-          bootstrap: get(config, 'Bootstrap'),
+        const libp2pOptions = {
+          peerInfo: self._peerInfo,
+          peerBook: self._peerInfoBook,
           modules: self._libp2pModules,
-          // EXPERIMENTAL
-          pubsub: get(self._options, 'EXPERIMENTAL.pubsub', false),
-          dht: get(self._options, 'EXPERIMENTAL.dht', false),
-          relay: {
-            enabled: get(self._options, 'EXPERIMENTAL.relay.enabled',
-              get(config, 'EXPERIMENTAL.relay.enabled', false)),
-            hop: {
-              enabled: get(self._options, 'EXPERIMENTAL.relay.hop.enabled',
-                get(config, 'EXPERIMENTAL.relay.hop.enabled', false)),
-              active: get(self._options, 'EXPERIMENTAL.relay.hop.active',
-                get(config, 'EXPERIMENTAL.relay.hop.active', false))
+          config: {
+            peerDiscovery: {
+              mdns: {
+                enabled: get(self._options, 'config.Discovery.MDNS.Enabled',
+                  get(config, 'Discovery.MDNS.Enabled', true))
+              },
+              webRTCStar: {
+                enabled: get(self._options, 'config.Discovery.webRTCStar.Enabled',
+                  get(config, 'Discovery.webRTCStar.Enabled', true))
+              },
+              bootstrap: {
+                list: get(self._options, 'config.Bootstrap',
+                  get(config, 'Bootstrap', []))
+              }
+            },
+            relay: {
+              enabled: get(self._options, 'EXPERIMENTAL.relay.enabled',
+                get(config, 'EXPERIMENTAL.relay.enabled', false)),
+              hop: {
+                enabled: get(self._options, 'EXPERIMENTAL.relay.hop.enabled',
+                  get(config, 'EXPERIMENTAL.relay.hop.enabled', false)),
+                active: get(self._options, 'EXPERIMENTAL.relay.hop.active',
+                  get(config, 'EXPERIMENTAL.relay.hop.active', false))
+              }
+            },
+            EXPERIMENTAL: {
+              dht: get(self._options, 'EXPERIMENTAL.dht', false),
+              pubsub: get(self._options, 'EXPERIMENTAL.pubsub', false)
```

Comment:
Nice! For the next turn, we should optimize js-ipfs options just like js-libp2p so that when we do `new IPFS(options)`, it is clear that:

```
const options = {
  ... // any specifics on spawning the daemon this way on the first level
  config: {
    // everything else comes from the config
  }
}
```